imod_coupler/drivers/dfm_metamod/msw_wrapper.py

```diff
@@ -0,0 +1,122 @@
+from typing import Optional
+
+import numpy as np
+from numpy.typing import NDArray
+from xmipy import XmiWrapper
+
+
+class MswWrapper(XmiWrapper):
+    def initiate_surface_water_component(self) -> None:
+
+        """
+        function to initialise the surface water component in metaswap.
+
+        Parameters
+        ----------
+        none
+
+        """
+        self.initiate_surface_water_component()
+
+    def start_surfacewater_timestep(self, idtsw: int) -> None:
+        """
+        function to start and excecute surface water timestep between msw and dflow
+
+        Parameters
+        ----------
+        idtsw : integer*4
+            index of time step within dtgw-cycle
+
+        """
+        self.start_surface_water_time_step(idtsw)
+
+    def finish_surfacewater_timestep(self, idtsw: int) -> None:
+        """
+        function to finish surface water timestep between msw and dflow
+
+        Parameters
+        ----------
+        idtsw : integer*4
+            index of time step within dtgw-cycle
+
+        """
+        self.finish_surface_water_time_step(idtsw)
+
+    def get_surfacewater_sprinking_demand(self) -> NDArray[np.float_]:
+        """returns the sprinkling volume demand from metaswap
+
+        Parameters
+        ----------
+        none
+
+        Returns
+        -------
+         NDArray[np.float_]:
+            sprinkling demand of MetaSWAP in m3/dtgw. Array as a copy of the MetaSWAP intenal array,
+            since the set function uses a different bmi/xmi-variable
+        """
+        sw_sprinkling_demand = self.get_value("dts2dfmputsp")
+        return sw_sprinkling_demand
+
+    def put_surfacewater_sprinking_demand(
+        self, sprinking_demand: NDArray[np.float_]
+    ) -> None:
+        """sets the sprinkling volume demand in metaswap
+
+        Parameters
+        ----------
+        sprinkiling_demand: NDArray[np.float_]:
+            sprinkling demand of MetaSWAP in m3/dtgw
+
+        Returns
+        -------
+        none
+
+        """
+        self.set_value("dts2dfmgetsp", sprinking_demand)
+
+    def get_surfacewater_ponding_allocation(self) -> NDArray[np.float_]:
+        """returns the ponding volume allocation from metaswap
+
+        Parameters
+        ----------
+        none
+
+        Returns
+        -------
+         NDArray[np.float_]:
+            ponding volume allocation of MetaSWAP in m3/dtsw. Array as a copy of the MetaSWAP intenal array,
+            since the set function uses a different bmi/xmi-variable
+        """
+        ponding_allocation = self.get_value("ts2dfmput")
+        return ponding_allocation
+
+    def put_surfacewater_ponding_allocation(
```

Comment:
" set" is more standard than " put"  so maybe "set_surfacewater_ponding_allocation"?